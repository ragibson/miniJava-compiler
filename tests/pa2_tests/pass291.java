// PA2 ref ASTs pass
class Refs {

    void p() {
        a = 1+2+3;
        a.b = 4;
        a.b.c = 5;
        this = that;
        this.that = 6;
        x = a[7];
    	a[8] = c;
        c.p(9);
        Foo x = a.b.c.d;
        Foo [] af = new Foo [10];
        that = new Test();
    }
}
