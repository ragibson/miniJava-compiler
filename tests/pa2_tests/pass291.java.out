********** miniJava Compiler (PA2) **********
Syntactic Analysis ... 
Syntactic Analysis OK
======= AST Display =========================
Package
  ClassDeclList [1]
  . ClassDecl
  .   "Refs" classname
  .   FieldDeclList [0]
  .   MethodDeclList [1]
  .   . (public) MethodDecl
  .   .   VOID BaseType
  .   .   "p" methodname
  .   .   ParameterDeclList [0]
  .   .   StmtList [11]
  .   .   . AssignStmt
  .   .   .   IdRef
  .   .   .     "a" Identifier
  .   .   .   BinaryExpr
  .   .   .     "+" Operator
  .   .   .       BinaryExpr
  .   .   .         "+" Operator
  .   .   .           LiteralExpr
  .   .   .             "1" IntLiteral
  .   .   .           LiteralExpr
  .   .   .             "2" IntLiteral
  .   .   .       LiteralExpr
  .   .   .         "3" IntLiteral
  .   .   . AssignStmt
  .   .   .   QualRef
  .   .   .     "b" Identifier
  .   .   .     IdRef
  .   .   .       "a" Identifier
  .   .   .   LiteralExpr
  .   .   .     "4" IntLiteral
  .   .   . AssignStmt
  .   .   .   QualRef
  .   .   .     "c" Identifier
  .   .   .     QualRef
  .   .   .       "b" Identifier
  .   .   .       IdRef
  .   .   .         "a" Identifier
  .   .   .   LiteralExpr
  .   .   .     "5" IntLiteral
  .   .   . AssignStmt
  .   .   .   ThisRef
  .   .   .   RefExpr
  .   .   .     IdRef
  .   .   .       "that" Identifier
  .   .   . AssignStmt
  .   .   .   QualRef
  .   .   .     "that" Identifier
  .   .   .     ThisRef
  .   .   .   LiteralExpr
  .   .   .     "6" IntLiteral
  .   .   . AssignStmt
  .   .   .   IdRef
  .   .   .     "x" Identifier
  .   .   .   RefExpr
  .   .   .     IxRef
  .   .   .       IdRef
  .   .   .         "a" Identifier
  .   .   .       LiteralExpr
  .   .   .         "7" IntLiteral
  .   .   . AssignStmt
  .   .   .   IxRef
  .   .   .     IdRef
  .   .   .       "a" Identifier
  .   .   .     LiteralExpr
  .   .   .       "8" IntLiteral
  .   .   .   RefExpr
  .   .   .     IdRef
  .   .   .       "c" Identifier
  .   .   . CallStmt
  .   .   .   QualRef
  .   .   .     "p" Identifier
  .   .   .     IdRef
  .   .   .       "c" Identifier
  .   .   .   ExprList [1]
  .   .   .   . LiteralExpr
  .   .   .   .   "9" IntLiteral
  .   .   . VarDeclStmt
  .   .   .   VarDecl
  .   .   .     ClassType
  .   .   .       "Foo" Identifier
  .   .   .     "x" varname
  .   .   .   RefExpr
  .   .   .     QualRef
  .   .   .       "d" Identifier
  .   .   .       QualRef
  .   .   .         "c" Identifier
  .   .   .         QualRef
  .   .   .           "b" Identifier
  .   .   .           IdRef
  .   .   .             "a" Identifier
  .   .   . VarDeclStmt
  .   .   .   VarDecl
  .   .   .     ArrayType
  .   .   .       ClassType
  .   .   .         "Foo" Identifier
  .   .   .     "af" varname
  .   .   .   NewArrayExpr
  .   .   .     ClassType
  .   .   .       "Foo" Identifier
  .   .   .     LiteralExpr
  .   .   .       "10" IntLiteral
  .   .   . AssignStmt
  .   .   .   IdRef
  .   .   .     "that" Identifier
  .   .   .   NewObjectExpr
  .   .   .     ClassType
  .   .   .       "Test" Identifier
=============================================
